File: angular/src/app/core/models/offre-emploi.ts
export interface OffreEmploi {
  id?: number;
  titre: string;
  description: string;
  entreprise: string;
  localisation: string;
  salaire: number;
  typeContrat: TypeContrat;
  addedBy  :number
}

export enum TypeContrat {
  CDD = 'CDD',
  CDI = 'CDI',
  FREELANCE = 'FREELANCE'
}

File: spring/src/main/java/feriel/offre/portail/entity/OffreEmploi.java
package feriel.offre.portail.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class OffreEmploi {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String titre;
    private String description;
    private String entreprise;
    private String localisation;
    private Double salaire;
    private TypeContrat typeContrat;
    private long addedBy ;
    @ManyToOne
    private Category category ;


}